tests/foreman/api/test_template.py

```diff
@@ -305,3 +305,349 @@ def test_positive_clone(self):
                 for key in uniqe_keys:
                     del new_template[key]
                 self.assertEqual(template_origin, new_template)
+
+
+class TemplateSyncTestCase(APITestCase):
+    """Implements TemplateSync tests from API"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Setup for TemplateSync functionality
+
+        :steps:
+
+            1. Git repository must exist (in gitlab or github) and its url
```

Comment:
some of these are not actual steps, but more preconditions, so I will put such into `:setup:` section